lib/operations/collection_ops.js

```diff
@@ -59,6 +77,98 @@ const loadDb = makeLazyLoader(`${__dirname}/../db`);
 const groupFunction =
   'function () {\nvar c = db[ns].find(condition);\nvar map = new Map();\nvar reduce_function = reduce;\n\nwhile (c.hasNext()) {\nvar obj = c.next();\nvar key = {};\n\nfor (var i = 0, len = keys.length; i < len; ++i) {\nvar k = keys[i];\nkey[k] = obj[k];\n}\n\nvar aggObj = map.get(key);\n\nif (aggObj == null) {\nvar newObj = Object.extend({}, key);\naggObj = Object.extend(newObj, initial);\nmap.put(key, aggObj);\n}\n\nreduce_function(obj, aggObj);\n}\n\nreturn { "result": map.values() };\n}';
 
+/**
+ * Perform an aggregate operation. See Collection.prototype.aggregate for more information.
+ *
+ * @method
+ * @param {Collection} a Collection instance.
+ * @param {object} [pipeline=[]] Array containing all the aggregation framework commands for the execution.
+ * @param {object} [options] Optional settings. See Collection.prototype.aggregate for a list of options.
+ * @param {Collection~aggregationCallback} callback The command result callback
+ */
+function aggregate(coll, pipeline, options, callback) {
```

Comment:
Refactoring this into a separate function is definitely necessary. However, this is not an "operation" in that we do not want to call `executeOperation` on it, and for most use cases (and the intended use cases) it is not async. It is more of a cursor factory function. This is a minor point, but I thought we should consider moving this to another file.